src/traces/histogram/calc.js

```diff
@@ -180,10 +157,181 @@ module.exports = function calc(gd, trace) {
     return cd;
 };
 
+/*
+ * calcAllAutoBins: we want all histograms on the same axes to share bin specs
+ * if they're grouped or stacked. If the user has explicitly specified differing
+ * bin specs, there's nothing we can do, but if possible we will try to use the
+ * smallest bins of any of the auto values for all histograms grouped/stacked
+ * together.
+ */
+function calcAllAutoBins(gd, trace, pa, maindata) {
+    var binAttr = maindata + 'bins';
+
+    // all but the first trace in this group has already been marked finished
+    // clear this flag, so next time we run calc we will run autobin again
+    if(trace._autoBinFinished) {
```

Comment:
> maybe `histogram/calc.js` should only sanitize `pos` and `size` values and call `arraysToCalcdata`, and leave the rest to `set_positions.js`?

Pro: that would avoid coupling between traces in a part of the code (`calc`) that's supposed to be operating on one trace at a time... and `set_positions` is only called after `calc` (as opposed to every replot), which is when it's needed.

Con: we'd need to be saving these sanitized `pos0` during `calc`, rather than making an actual `calcdata[i]` array - or perhaps stashing it in `calcdata[i][0]` as we used to do with trace-wide stuff (and still do here and there) but then `setPositions` would need to fill in the rest of `calcdata[i]` once it determined the bin spec.

So I feel like in the end it would be more confusing that way. Thoughts?

Incidentally, it looks like we're probably calling `setPositions` twice unnecessarily if we have both histogram and bar traces on the same plot. I'll investigate...